x509/cert_validations.go

```diff
@@ -128,24 +129,24 @@ func validateSigningTime(cert *x509.Certificate, signingTime *time.Time) error {
 	return nil
 }
 
-func validateCACertificate(cert *x509.Certificate, expectedPathLen int) error {
+func validateCACertificate(cert *x509.Certificate, expectedPathLen int, isTimestamp bool) error {
```

Comment:
It seems that this function cannot be reused well for timestamping, especially for the function `validateExtendedKeyUsage`. Could you create a new file just for TSA cert validation?